pkg/operators/oci-handler/oci.go

```diff
@@ -264,6 +266,91 @@ func (o *OciHandlerInstance) init(gadgetCtx operators.GadgetContext) error {
 		return fmt.Errorf("unmarshalling metadata: %w", err)
 	}
 
+	// Extract custom params
+	customParams := viper.Sub("params.custom")
+	if customParams != nil {
+		for k := range viper.GetStringMap("params.custom") {
+			log.Debugf("evaluating custom param %q", k)
+			paramSub := customParams.Sub(k)
+			valuesSub := paramSub.Sub("values")
+			if valuesSub == nil {
+				log.Debugf("custom param %q has no values", k)
+				continue
+			}
+			p := &api.Param{
+				Key:          k,
+				Description:  paramSub.GetString("description"),
+				Prefix:       "custom.",
+				DefaultValue: paramSub.GetString("defaultValue"),
+			}
+			for value := range paramSub.GetStringMap("values") {
+				// Skip wildcard param
+				if value == "*" {
+					continue
+				}
+				p.PossibleValues = append(p.PossibleValues, value)
+			}
+			o.extraParams = append(o.extraParams, p)
+
+			// Evaluate, if set
+			if val, ok := o.paramValues["custom."+k]; ok {
+				valName := val
+				if !paramSub.IsSet("values."+valName+".applyConfig") && paramSub.IsSet("values.*.applyConfig") {
+					// Use wildcard fallback
+					valName = "*"
+				}
+
+				log.Debugf("applying custom param %q value %q", k, valName)
+
+				valSub := paramSub.Sub("values." + valName + ".applyConfig")
```

Comment:
With the current code, the `applyConfig` section only works when the parameter is in the "custom" group (and not when it is in the "ebpf" group). Shouldn't it be made more generic so that parameters belonging to the ebpf operator can also apply some config changes? (It would require something like the Owner field mentioned above).

Reply:
I'm not sure about allowing this behavior also in other sections. I'd rather have the option to hide ebpf (or other) params and then still set their values using the custom params (`--print-ustack` (custom param) setting `--collect-ustack` (ebpf param), even if hidden - I know this example doesn't make sense 😄).

Otherwise, I think this would become even more complex than it already is.